src/mapgen.cpp

```diff
@@ -2429,18 +2436,21 @@ class jmapgen_monster : public jmapgen_piece
             }
 
             mongroup_id chosen_group = m_id.get( dat );
-            std::string chosen_name = name.translated();
+            std::optional<std::string> chosen_name = std::nullopt;
             if( !random_name_str.empty() ) {
                 if( random_name_str == "female" ) {
-                    chosen_name = SNIPPET.expand( "<female_given_name>" );
+                    chosen_name.emplace( SNIPPET.expand( "<female_given_name>" ) );
                 } else if( random_name_str == "male" ) {
-                    chosen_name = SNIPPET.expand( "<male_given_name>" );
+                    chosen_name.emplace( SNIPPET.expand( "<male_given_name>" ) );
                 } else if( random_name_str == "random" ) {
-                    chosen_name = SNIPPET.expand( "<given_name>" );
+                    chosen_name.emplace( SNIPPET.expand( "<given_name>" ) );
                 } else if( random_name_str == "snippet" ) {
-                    chosen_name = SNIPPET.expand( name.translated() );
+                    chosen_name.emplace( SNIPPET.expand( random_name_snippet_id ) );
```

Comment:
Previously this allowed strings with snippet tags be used (such as `Mr. <given_name>`), and the string extraction script also extracts this string, so I think we should keep this behavior.